src/main/java/com/khoros/twitterapp/services/TwitterService.java

```diff
@@ -76,55 +73,87 @@ public TwitterService(Twitter twitterFactory) {
 
     public Optional<List<Status>> getHomeTimeline() throws TwitterServiceException {
 
-        return getHomeTimelineFilteredByKeyword(null);
+        return getFilteredHomeTimeline(null);
 
     }
 
-    public Optional<List<Status>> getHomeTimelineFilteredByKeyword(String keyword) throws TwitterServiceException {
+    public Optional<List<Status>> getFilteredHomeTimeline(String keyword) throws TwitterServiceException {
 
         logger.info("Attempting to retrieve home timeline through Twitter API.");
 
-        Set<twitter4j.Status> cacheSet = cacheUp.getCacheSet();
         Optional<List<twitter4j.Status>> optionalList = null;
 
-        if(cacheSet.isEmpty()) {
+        if(cacheUp.getTimelineCache().containsKey(CacheListType.HOME)) {
+
+            optionalList = Optional.ofNullable(cacheUp.getTimelineCache().get(CacheListType.HOME));
+
+        } else {
 
             try {
 
                 optionalList = Optional.ofNullable(twitterFactory.getHomeTimeline());
-                optionalList.ifPresent((list) -> cacheUp.addStatusesToCache(list));
+                optionalList.ifPresent(list -> cacheUp.addStatusToCache(CacheListType.HOME, list));
 
             } catch (TwitterException twitterException) {
 
-                logger.info("Timeline retrieval aborted. Twitter Exception thrown.");
+                logger.info("Home timeline retrieval aborted. Twitter Exception thrown.");
 
                 handleTwitterException(twitterException);
 
             }
 
-        } else {
-
-            List<twitter4j.Status> responseList = new ArrayList<>();
-            responseList.addAll(cacheSet);
-            optionalList = Optional.ofNullable(responseList);
-
         }
 
         return optionalList
                 .map(list -> list.stream()
-                      .filter(originalStatus -> {
+                        .filter(originalStatus -> {
+
+                            if (StringUtils.isEmpty(keyword)) {
+
+                                return true;
+
+                            } else {
+
+                                return originalStatus.getText().contains(keyword);
+                            }
+                        })
+                        .map(thisStatus -> createNewStatusObject(thisStatus))
+                        .collect(Collectors.toList())
+                );
+
+    }
+
+    public Optional<List<Status>> getUserTimeline() throws TwitterServiceException {
+
+        logger.info("Attempting to retrieve user timeline through Twitter API.");
 
-                          if (StringUtils.isEmpty(keyword)) {
+        Optional<List<twitter4j.Status>> optionalList = null;
+
+        if(cacheUp.getTimelineCache().containsKey(CacheListType.USER)) {
```

Comment:
this works, if you wanted top optimize it you could pull it directly check if it's null and then pull from twitter